transportable-udfs-codegen/src/main/java/com/linkedin/transport/codegen/SparkWrapperGenerator.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.codegen;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+import com.squareup.javapoet.ClassName;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.text.StringSubstitutor;
+
+
+public class SparkWrapperGenerator implements WrapperGenerator {
+
+  private static final String SPARK_PACKAGE_SUFFIX = "spark";
+  private static final String SPARK_WRAPPER_TEMPLATE_RESOURCE_PATH = "wrapper-templates/spark";
+  private static final String SUBSTITUTOR_KEY_WRAPPER_PACKAGE = "wrapperPackage";
+  private static final String SUBSTITUTOR_KEY_WRAPPER_CLASS = "wrapperClass";
+  private static final String SUBSTITUTOR_KEY_UDF_TOP_LEVEL_CLASS = "udfTopLevelClass";
+  private static final String SUBSTITUTOR_KEY_UDF_IMPLEMENTATIONS = "udfImplementations";
+
+  @Override
+  public void generateWrappers(ProjectContext context) {
+    Multimap<String, String> udfs = context.getUdfProperties().getUdfs();
```

Comment:
didn't fully follow how this works - does the developer have to list all the functions in a json file? If yes, can they simply annotate the code and we scan the package for annotations instead of separate json file